build/xtask/src/dist.rs

```diff
@@ -1287,6 +1287,7 @@ fn link(
     cmd.arg("-m").arg(m);
     cmd.arg("-z").arg("common-page-size=0x20");
     cmd.arg("-z").arg("max-page-size=0x20");
+    cmd.arg("-rustc-lld-flavor=ld");
```

Comment:
FWIW: I don't know why I need to make this change. In theory this is supposed to be detected, but isn't now? The docs on this are effectively non-existent...

Reply:
A brief recap about what led to needing to provide `-rustc-lld-flavor`:

Calling `ld` directly to link is a bit of a pain as you need to deal with all the platform specific logic (e.g. like where to find crt1.o) which you can avoid by invoking it via `cc/gcc/clang`. Up until recently, you couldn't pass the linker via an absolute path to `gcc` but instead had to add a search path that would have the `ld` binary. Hence why `$SYSROOT/lib/rustlib/$HOST_TARGET/bin/gcc-ld` exists. This is what happens when you pass `-Zgcc-ld=lld` (soon to be stabilized as `-Clink-self-contained=linker -Clinker-flavor=gcc-lld`) to `rustc` to use the bundled `rust-lld`.

Now, `lld` decides what flavour to use based on either the name of the binary invoked or via the first argument (i.e., `arg[0]` or `arg[1]`). Unfortunately, rust couldn't just ship `ld` and `ld64` as symlinks to `rust-lld` so the `gcc-ld` folder just had multiple copies which meant some bloat.

All's fine for Hubris at this point because we just directly invoked `$SYSROOT/lib/rustlib/$HOST_TARGET/bin/gcc-ld/ld` and so `lld` operated with the `ld` flavour.

Around Oct last year to reduce the bloat (3 copies of `rust-lld`), the binaries under `gcc-ld` were replaced with a small tool `lld-wrapper` that would call `../rust-lld` with the right flavor. So now the `ld` and `ld64` under `bin/gcc-ld/` were `lld-wrapper` compiled with different `cfg` flags choosing the flavour to pass to `../rust-lld`. Up until this point, I'd expect `hubris` to still work when invoking this `ld`.

But this was still a pain, with the duplication and lacked some other support. So in May-ish, it was simplified to a single copy that would correctly place the flavour argument as the first one passed to `rust-lld` via `-rustc-lld-flavor`. Although, looks like this did come with some [downsides](https://github.com/rust-lang/rust/issues/97755) that might mean further changes coming.

Looking at hubris, looks like we're manually invoking the linker here and not through `cc/gcc/clang`. I would say we should just call `rust-lld` directly instead of using whatever is in under `bin/gcc-ld` (really all that is an implementation detail for `rustc` hence the name of the flag).